lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -398,6 +398,18 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
 //===----------------------------------------------------------------------===//
 
 namespace {
+struct ModuleEmitterOptions {
+  /// True if always_ff blocks should not be emitted.
+  bool noAlwaysFF = false;
+};
+
+/// Read the emitter options out of the module op attributes and store them in
+/// the emitter state.
+static void loadModuleOptions(ModuleOp module,
+                              ModuleEmitterOptions &moduleOptions) {
+  moduleOptions.noAlwaysFF = module->hasAttr("sv.no_alwaysff");
```

Comment:
I'd recommend making this a ctor on `ModuleEmitterOptions` that takes a `StringRef`, and a closure to call back for errors.